src/ui/components/SecondaryToolbox/ReactDevTools.tsx

```diff
@@ -261,7 +263,7 @@ class ReplayWall implements Wall {
       assert(this.pauseId, "Must have a pause ID to handle a response!");
       const result: any = await getJSON(this.replayClient, this.pauseId, response.returned);
 
-      if (result) {
+      if (result && this.pauseId === originalPauseId) {
```

Comment:
This (race condition type thing) looks like a good thing for us to add test coverage for, though I'm not sure how easy it would be to do.